libs/internal/include/launchdarkly/network/asio_requester.hpp

```diff
@@ -68,29 +65,49 @@ static http::verb ConvertMethod(HttpMethod method) {
     launchdarkly::detail::unreachable();
 }
 
+static std::string HostWithOptionalPort(
+    std::string host,
+    std::optional<std::string> const& port) {
+    if (!port || (*port == "80" || *port == "443")) {
+        return host;
+    }
+    host.append(":");
+    host.append(*port);
+    return host;
+}
+
 static http::request<http::string_body> MakeBeastRequest(
     HttpRequest const& request) {
     http::request<http::string_body> beast_request;
 
-    beast_request.method(ConvertMethod(request.Method()));
-    auto body = request.Body();
-    if (body) {
+    if (auto const body = request.Body()) {
         beast_request.body() = body.value();
     } else {
         beast_request.body() = "";
     }
+
+    beast_request.prepare_payload();
+
+    beast_request.method(ConvertMethod(request.Method()));
+
     if (request.Path().empty()) {
         beast_request.target("/");
     } else {
         beast_request.target(request.Path());
     }
 
-    beast_request.prepare_payload();
-    beast_request.set(http::field::host, request.Host());
+    // If we have an HTTP proxy set, we need to pass the entire path
+    // in the HTTP request's target (so the proxy knows where to send it).
+    if (request.HttpProxyHost()) {
+        beast_request.target(request.Url());
+    }
+
+    beast_request.set(http::field::host,
+                      HostWithOptionalPort(request.Host(), request.Port()));
 
     auto const& properties = request.Properties();
 
-    for (auto const& pair : request.Properties().BaseHeaders()) {
+    for (auto const& pair : properties.BaseHeaders()) {
```

Comment:
This variable was unused before. I think it was there fore readability, so I updated the code to use it.